pkg/virt-controller/watch/migration/migration_test.go

```diff
@@ -1296,6 +1298,108 @@ var _ = Describe("Migration watcher", func() {
 			),
 		)
 
+		It("should update network-info annotation for SRIOV interface during handoff", func() {
+			vmi := newVirtualMachine("testvmi", virtv1.Running)
+			addNodeNameToVMI(vmi, "node02")
+
+			vmi.Spec.Domain.Devices.Interfaces = []virtv1.Interface{
+				{
+					Name:                   "sriov-net",
+					InterfaceBindingMethod: virtv1.InterfaceBindingMethod{SRIOV: &virtv1.InterfaceSRIOV{}},
+				},
+			}
+			vmi.Spec.Networks = []virtv1.Network{
+				{Name: "sriov-net", NetworkSource: virtv1.NetworkSource{Multus: &virtv1.MultusNetwork{NetworkName: "sriov-network"}}},
+			}
+
+			migration := newMigration("testmigration", vmi.Name, virtv1.MigrationScheduled)
+			targetPod := newTargetPodForVirtualMachine(vmi, migration, k8sv1.PodRunning)
+			targetPod.Spec.NodeName = "node01"
+			targetPod.Status.ContainerStatuses = []k8sv1.ContainerStatus{{
+				Name: "compute", State: k8sv1.ContainerState{Running: &k8sv1.ContainerStateRunning{}},
+			}}
+
+			targetPod.Annotations = map[string]string{
+				"k8s.v1.cni.cncf.io/network-status": `[{"name":"sriov-network","interface":"net1","device-info":{"type":"pci","version":"1.0.0","pci":{"pci-address":"0000:01:00.0"}}}]`,
+			}
+
+			expectedNetworkInfo := `{"interfaces":[{"network":"sriov-net","deviceInfo":{"type":"pci","version":"1.0.0","pci":{"pci-address":"0000:01:00.0"}}}]}`
+			controller.netAnnotationsGenerator = stubNetworkAnnotationsGenerator{
+				annotations: map[string]string{downwardapi.NetworkInfoAnnot: expectedNetworkInfo},
+			}
+
+			addMigration(migration)
+			addVirtualMachineInstance(vmi)
+			addPod(newSourcePodForVirtualMachine(vmi))
+			addPod(targetPod)
+
+			sanityExecute()
+
+			updatedPod, err := kubeClient.CoreV1().Pods(targetPod.Namespace).Get(context.Background(), targetPod.Name, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(updatedPod.Annotations).To(HaveKey(downwardapi.NetworkInfoAnnot))
+			Expect(updatedPod.Annotations[downwardapi.NetworkInfoAnnot]).To(Equal(expectedNetworkInfo))
+
+			testutils.ExpectEvent(recorder, virtcontroller.SuccessfulHandOverPodReason)
+		})
+
+		It("should update network-info annotation for binding plugin interface during handoff", func() {
+			setConfig(&virtv1.KubeVirtConfiguration{
+				NetworkConfiguration: &virtv1.NetworkConfiguration{
+					Binding: map[string]virtv1.InterfaceBindingPlugin{
+						"test-plugin": {
+							DownwardAPI: virtv1.DeviceInfo,
+						},
+					},
+				},
+			})
```

Comment:
Since the migration controller accepts an interface for the network annotation generator, it is sufficient to just pass in the `stubNetworkAnnotationsGenerator`. The real cluster config is unnecessary because the stub controls the expected output directly.